sklearn/decomposition/sparse_pca.py

```diff
@@ -185,6 +191,63 @@ def transform(self, X, ridge_alpha='deprecated'):
         U /= s
         return U
 
+    def get_explained_variance(self, components, X):
+        '''
```

Comment:
Do we want this function to be used from outside ?
otherwise I think it should be private: `_get_explained_variance`

still in the optics to be a private function, you don't need to pass components to the function. You can directly access self.components_ inside the function.

Reply:
renamed thanks!